components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/util/Utils.java

```diff
@@ -187,6 +189,31 @@ public static NotFoundException buildNotFoundException(String description, Strin
         return new NotFoundException(errorDTO);
     }
 
+    /**
+     * Check whether correlation id present in the log MDC
+     *
+     * @return whether the correlation id is present
+     */
+    public static boolean isCorrelationIDPresent() {
+        return MDC.get(Constants.CORRELATION_ID_MDC) != null;
+    }
+
+    /**
+     * Get correlation id of current thread
+     *
+     * @return correlation-id
+     */
+    public static String getCorrelation() {
+        String ref;
+        if (isCorrelationIDPresent()) {
+            ref = MDC.get(Constants.CORRELATION_ID_MDC).toString();
+        } else {
+            ref = UUID.randomUUID().toString();
```

Comment:
don't we need to add the generated id to MDC if not present?

Reply:
Fixed with 54e80535d2c691cd16bbc38ee1548f83cab98dda